### PR TITLE
Functional tests

### LINKED ISSUE #45: Make uberjar
So we are able to use it standalone. Such jar should be not used as a thirdparty dependency.

https://github.com/sbt/sbt-assembly

### LINKED ISSUE #55: Functional tests
### Pieces
- [ ] #231 Funtests Windows support
- [ ] #232 REPL funtests support
### Description

We have two modes right now:
- REPL
- Execution

The scope of this issue is to create a bunch of Morganey scripts and programs that cover all of its features, run them on CI and check if everything acts as expected.

Those test should be easily run locally on the developers machines to check stuff before pushing it.

### PR DESCRIPTION
**Pull request by rexim**
### Description

Since I'm really bad at sbt, my funtests infrastructure implementation probably could be better. Please provide your remarks when you get a chance. 
### Split out pieces
- Funtests Windows support
- REPL funtests support
### Checklist
- [x] Update docs
- [x] Add/Update Unit Tests

---

Split #55

### CODE REVIEW

**Comment by rexim** [comment]
Alright, looks funtests hang on Windows. I'm gonna fix that later.

**Comment by ForNeVeR** [comment]
> looks funtests hang on Windows. I'm gonna fix that later.

Feel free to ask me if you need any help.

**Comment by rexim** [comment]
> Feel free to ask me if you need any help.

@ForNeVeR sure! Thank you!

**Comment by rexim** [comment]
Alright, it's been already more than a week. I'm gonna submit what I've already done and split out the Windows funtests support.

**Comment by rexim** [comment]
This is interesting. The result of REPL funtests heavily depends on the terminal implementation on Linux. I observed different results on gnome-terminal and Emacs. It's probably caused by jline. I'm gonna split out REPL funtests support too.

**Comment by rexim** [comment]
@ForNeVeR @keddelzz PR is ready. Please take a look when you get a chance.

File: funtests/src/test/scala/me/rexim/morganey/funtests/FibFuntest.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.io.File
+
+import org.scalatest._
+import scala.sys.process._
+
+class FibFuntest extends FlatSpec with Matchers {
+  "15 first fibonacci numbers sample" should "print 15 first fibonacci numbers" in {
+    val expectedOutput = "numbers: [0,1,1,2,3,5,8,13,21,34,55,89,144,233,377]\n"
+    val actualOutput = "java -jar ./target/scala-2.11/morganey.jar ./docs/samples/fib.mgn".!!
```
Comment (ForNeVeR) [inline funtests/src/test/scala/me/rexim/morganey/funtests/FibFuntest.scala]: I recommend to factor this out into a function or a constant, to remove the duplication of `"java -jar ./target/scala-2.11/morganey.jar"` everywhere.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Review by keddelzz — APPROVED** [review]
Besides @ForNeVeR's suggested improvement; LGTM :+1:

File: funtests/src/test/scala/me/rexim/morganey/funtests/FibFuntest.scala (hunk 1 above)
Comment (rexim) [reply funtests/src/test/scala/me/rexim/morganey/funtests/FibFuntest.scala]: Done

**Review by ForNeVeR — APPROVED** [review]

**Comment by rexim** [comment]
@ForNeVeR @keddelzz thanks for the review guys!!

**Pull request merged**